agent/util-scripts/pbench-tool-meister-start

```diff
@@ -731,13 +733,86 @@ def main(_prog: str, cli_params: Namespace) -> int:
+        # us connecting, and we'll use that (by default) as the server address.
+        localhost = LocalRemoteHost()
+        origin_ip = set()
+        template = TemplateSsh(ssh_cmd, shlex.split(ssh_opts), ['echo', '${SSH_CONNECTION}'])
+        for host in tool_group.hostnames.keys():
+            if not localhost.is_local(host):
+                template.start(host)
+
+        for host, params in tool_group.hostnames.items():
+            if not localhost.is_local(host):
+                connection = template.wait(host)
+                logger.debug("Host %s reports connection `%s`", host, connection)
+                if connection[0] == 0 and connection[1]:
+                    # The SSH_CONNECTION value is the full `stdout` and we
+                    # don't expect a stderr on success; the format is
+                    #   "origin_node origin_port local_node local_port"
+                    # we only need the origin node because we know that's an
+                    # IP for our local TDS and Redis host that the remote can
+                    # understand.
+                    origin = connection[1].split()[0]
+                    origin_ip.add(origin)
+
+                    # Store the origin reported by this remote in its parameter
+                    # block: we'll program this as the connection address when
+                    # we send the init handshake to that remote later.
+                    params["origin_host"] = origin
+                else:
+                    # If `ssh` fails, there's no point in continuing!
+                    logger.error("Host %s reports %s", host, connection)
+                    return ReturnCode.REMOTENOTREACHABLE
+
+        # Convert the de-duplicating set into an indexable list. Ideally there
+        # will be only a single entry here, since the current BaseServer init
+        # (and the Bottle WSGI server) doesn't support listening on multiple
+        # addresses; however we're going to just take the first returned origin
+        # address and hope they can all connect.
+        #
+        # NOTE: an alternative would be (in the absence of an explicit
+        # connection spec) to always listen on '0.0.0.0', but this eliminates
+        # IPv6, and we currently lack the mechanism to listen on both '0.0.0.0'
+        # and '::' (the IPv6 equivalent, aka '0:0:0:0:0:0:0:0').
+        origin_list = list(origin_ip)
+
+        # NOTE: If the caller/human supplied explicit PBENCH_REDIS_SERVER or
+        # PBENCH_TOOL_DATA_SINK (--redis-server or --tool-data-sink), we'll use
+        # those, but by default we'll use the ssh connection origin instead of
+        # the local `hostname` which may not be reachable by the remotes (or
+        # necessarily routable at all).
+        #
+        # If we aren't able to contact any remotes, or if there were no remotes
+        # in the first place, then the origin_list will be empty, and we'll
```

Comment:
Actually, if there were any remotes which we were unable to contact, then control would have returned by now and `origin_list` would never have been created...right?